myfitnesspal/client.py

```diff
@@ -795,3 +795,127 @@ def get_food_item_details(self, mfp_id: int) -> FoodItem:
+
+        # Step4 - Build Post Data and finally submit new Food with nutritional Details
+        data = {
+            "utf8": utf8_field,
+            "authenticity_token": authenticity_token,
+            "date": datetime.datetime.today().strftime("%Y-%m-%d"),
+            "food[brand]": brand,
+            "food[description]": description,
+            "weight[serving_size]": serving_size,
+            "servingspercontainer": "{}".format(servingspercontainer),
+            "nutritional_content[calories]": "{}".format(calories),
+            "nutritional_content[sodium]": "{}".format(sodium),
+            "nutritional_content[fat]": "{}".format(fat),
+            "nutritional_content[potassium]": "{}".format(potassium),
+            "nutritional_content[saturated_fat]": "{}".format(saturated_fat),
+            "nutritional_content[carbs]": "{}".format(carbs),
+            "nutritional_content[polyunsaturated_fat]": "{}".format(polyunsaturated_fat),
+            "nutritional_content[fiber]": "{}".format(fiber),
+            "nutritional_content[monounsaturated_fat]": "{}".format(monounsaturated_fat),
+            "nutritional_content[sugar]": "{}".format(sugar),
+            "nutritional_content[trans_fat]": "{}".format(trans_fat),
+            "nutritional_content[protein]": "{}".format(protein),
+            "nutritional_content[cholesterol]": "{}".format(cholesterol),
+            "nutritional_content[vitamin_a]": "{}".format(vitamin_a),
+            "nutritional_content[calcium]": "{}".format(calcium),
+            "nutritional_content[vitamin_c]": "{}".format(vitamin_c),
+            "nutritional_content[iron]": "{}".format(iron),
+            "food_entry[quantity]": "1.0",
+            "food_entry[meal_id]": "0",
+            "addtodiary": "no",
+            "preserve_exact_description_and_brand": "true",
+            "continue": "Save"
+        }
+        # Make entry public if requested, Hint: submit "sharefood": 0 also generates a public db entry, so only add
+        # "sharefood"" if really requested
+        if sharepublic:
+            data["sharefood"] = 1
+
+        submit4_url = parse.urljoin(self.BASE_URL_SECURE, self.SUBMIT_POST_PATH)
+        result = self.session.post(submit4_url, data, )
+        if result.status_code == 200:
+            document = lxml.html.document_fromstring(result.content.decode('utf-8'))
+            try:
+                if not document.xpath(
+                        # If list is empty there should be no error, could be replaced with assert
+                        "//*[@id='errorExplanation']/ul/li"):
+                    return True
+                    # print("No Error :)")
+                    # Would like to return FoodItem, but seems that it take
+                    # to long until the submitted food is available in the DB
+                    # return self.get_food_search_results("{} {}".format(brand, description))[0]
```

Comment:
Yeah, hmm, this looks tough. In some ways it's not entirely our job to fix their read-after-write inconsistency, so I'm not sure we should try to hack it on our end. And IMO, we probably don't need to return a `FoodItem`, since within the web application, the UX doesn't seem to be showing the created item after it is created—it just redirects me back to my diary. So we can probably just `return` here without worrying about the get food search results call.

(But we can and should defer to the maintainers. This is just my take.)

Reply:
I agree with you. Was just an idea for a nice to have feature :)